src/ResourceManagement/ResourceManager/Authentication/MSITokenProvider.cs

```diff
@@ -45,7 +53,49 @@ public MSITokenProvider(string resource, MSILoginInformation msiLoginInformation
 
         private async Task<AuthenticationHeaderValue> GetAuthenticationHeaderForVirtualMachineAsync(string resource, CancellationToken cancellationToken = default(CancellationToken))
         {
-            int port = msiLoginInformation.Port == null ? 50342 : msiLoginInformation.Port.Value;
+            if (msiLoginInformation.Port != null)
+            {
+                // Token retrival from VM extension will be removed in the next v1.10 release as IMDS service
+                // replaces VM extension. MsiLoginInformation.Port is marked as deprecated and will be removed
+                //
+                var token = await GetTokenFromMSIExtensionAsync(msiLoginInformation.Port.Value, 
+                    resource, 
+                    cancellationToken);
+                return new AuthenticationHeaderValue(token.TokenType, token.AccessToken);
+            }
+            else
+            {
+                var token = await GetTokenFromIMDSEndpointAsync(resource, cancellationToken);
+                return new AuthenticationHeaderValue(token.TokenType, token.AccessToken);
+            }
+        }
+
+        private async Task<AuthenticationHeaderValue> GetAuthenticationHeaderForAppServiceAsync(string resource, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var endpoint = Environment.GetEnvironmentVariable("MSI_ENDPOINT") ?? throw new ArgumentNullException("MSI_ENDPOINT");
+            var secret = Environment.GetEnvironmentVariable("MSI_SECRET") ?? throw new ArgumentNullException("MSI_SECRET");
+            HttpRequestMessage msiRequest = new HttpRequestMessage(HttpMethod.Get, $"{endpoint}?resource={resource}&api-version=2017-09-01");
```

Comment:
>2017-09-01 [](start = 128, length = 10)

please extract the version to a constant and put it at the top of the file. It would be way much easier to upgrade and check what version we are using.